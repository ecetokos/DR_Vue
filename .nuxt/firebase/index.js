import createApp from './app.js'

import authService from './service.auth.js'

const appConfig = {"apiKey":"AIzaSyD0c9a38G_8L7k09tqHmrmU2Hk_CeUBhEQ","authDomain":"dr-firebase-9de85.firebaseapp.com","projectId":"dr-firebase-9de85","storageBucket":"dr-firebase-9de85.appspot.com","messagingSenderId":"602632802158","appId":"1:602632802158:web:f59e6d568054f7f743fb7f"}

export default async (ctx, inject) => {
  /****************************************
  **************** LAZY MODE **************
  ****************************************/

  const { firebase, session } = await createApp(appConfig, ctx)

  let servicePromises = []

  if (process.server) {
    servicePromises = [
      authService(session, firebase, ctx, inject),

    ]
  }

  if (process.client) {
    servicePromises = [
      authService(session, firebase, ctx, inject),

    ]
  }

  const [
    auth
  ] = await Promise.all(servicePromises)

  const fire = {
    auth: auth
  }

    inject('fireModule', firebase)
    ctx.$fireModule = firebase

  inject('fire', fire)
  ctx.$fire = fire
}
