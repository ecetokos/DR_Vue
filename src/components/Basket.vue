<template>
    <div class="container" style="margin-top:40px">
                    <div class="row">
                            <div class="container">
    </div>     
                        <div class="col-xs-12 subHeader blackColor">
                            <h1> Sepetim </h1> <span>( {{basketitems.length}} Ürün )</span>
                        </div>

                        <div class="basketContentFrame">

                            <div class="basketLeftFrame col-md-7 col-lg-8" style="float:left;">

                                <div class="basketProductSummary">

                                    <div class="basketHead row col-hidden">
                                        <div class="col-1 col-sm-1">
                                            <label class="control control-checkbox">
                                                <input type="checkbox" id="checkall" class="newChk" />
                                                <span class="control-indicator"></span>
                                            </label>
                                        </div>
                                        <div class="col-11 col-sm-5">ÜRÜN </div>
                                        <div class="col-sm-1"> &nbsp; </div>
                                        <div class="col-sm-2">ADET </div>
                                        <div class="col-sm-3 text-right">FİYAT </div>
                                    </div>
                                    <ul class="urunler">
    <li v-for="item in basketitems" v-bind:key="item">
        <div class="basketRow row" id="113409924" data-id="113409924"> 
        <input type="hidden" id="basketProductName" value="Son C&#252;ret">
        <input type="hidden" id="basketItemManufacturer" value="">
        <input type="hidden" id="basketItemCategory" value="">
        <input type="hidden" id="basketPersonProduct" value="Yılmaz &#214;zdil">
        <input type="hidden" id="basketItemPrice" value="25,2">
        <input type="hidden" id="basketItemMediaType">
        <div class="col-1 col-sm-1 firstColumn col-hidden">
            <div class="cell">
                <label class="control control-checkbox">
                    <input class="basketChk newChk productItemChk" type="checkbox" value="113409924" />
                    <span class="control-indicator"></span>
                </label>
            </div>
         </div>
                                            <div class="col-8 col-sm-6 secondColumn">
                                                <div class="cell">
                                                    <div class="basketImg"><a href="/Kitap/Son-Curet/Yilmaz-Ozdil/Arastirma-Tarih/Politika-Arastirma/Turkiye-Politika-/urunno=0001889645001"><img :src="item.img" alt="" /></a></div>
                                                    <div class="basketProductInfo">
                                                        <div class="name product-name" data-id="113409924" data-q="1" data-pid="1251728">
                                                            <a href="/Kitap/Son-Curet/Yilmaz-Ozdil/Arastirma-Tarih/Politika-Arastirma/Turkiye-Politika-/urunno=0001889645001">{{item.name}}</a>
                                                        </div>
                                                        <div class="author">{{item.author_name}}</div>
                                                        <div class="mediaType"></div>

                                                    </div>
                                                </div>
                                            </div>

                                            <div class="col-4 col-sm-5 flexCol">
                                                <div class="row flex xs-flex-row xs-height100">
                                                    <div class="col-12 col-sm-6 flexOrderLast">
                                                        <div class="cell xs-flex-end">
                                                            <div class="quantityGroup  ">
                                                                <div class="input-counter"><span class="number-minus disabled" @click="Decrase()"></span><input type="text" id="qty-113409924" class="basketCounter" readonly="" :value="count" step="1" min="1" max="50"><span class="number-plus"  @click="count += 1"></span></div>
                                                                    <div class="qupdate"> <a href="javascript:;" class="updateQuantity" data="113409924"> Güncelle </a> </div>
                                                            </div>
                                                        </div>
                                                    </div>
                                                    <div class="col-12 col-sm-6 text-right lastColumn">
                                                        <div class="cell">
                                                                <div class="discountGroup">
                                                                    <span>%{{item.discount}} indirim</span>
                                                                    <span>{{item.price_w_d}} TL</span>
                                                                </div>
                                                            
                                                            <div class="basketPrice"> {{(item.price)*count}} TL</div>
                                                            <div class="actionGroup">
                                                                <a href="javascript:;" class="actionLink addFav" onclick="addFavoriteShoppingCart(113409924);">Favorilere Ekle</a>
                                                                <span class="tooltipBtn right xs-left" data-tooltip="Favorilerinize Hesabım ekranından ulaşabilirsiniz."><img src="https://www.flaticon.com/svg/static/icons/svg/864/864381.svg" height="15px" width="15px"/></span>
                                                                <a href="javascript:;" class="actionLink" onclick="deleteShoppingCart(113409924);" style="margin-left:7px">Sil</a>
                                                            </div>
                                                        </div>
                                                    </div>
                                                </div>

                                            </div>
   </div> 
    </li>
</ul>
                                </div>
                                <!-- basketProductSummary End *******-->
<div class="giftPackage">
    <div class="chooseRow">
        <img src="https://www.flaticon.com/svg/static/icons/svg/1161/1161667.svg" height="20px" width="20px" style="margin-right:15px"/>
        <span class="gift-text"> Hediye Paketi İstiyorum</span>
        <span class="editGiftText">Düzenle</span>
        <span id="removeGiftText" class="removeGiftText">Sil</span>
        <div class="editableGiftText">
            <strong>Hediye Notu:</strong>
            <span id="displayGiftMessage"></span>
        </div>
    </div>
</div>
                                <!-- giftPackage end -->
                                <div class="totalActions col-hidden">
                                    <a class="totalRemove" href="javascript:;"> Sil</a>
                                    <a class="totalAddFav" href="javascript:;"> Favorilerime Ekle</a>
                                </div>
                                <!-- totalActions end -->
                            </div>
                            <!--**** basketLeftFrame End ****-->
                            <div class="basketRightFrame col-md-5 col-lg-4" style="width:400px; height:300px">
                                <div class="checkoutCartDetails">

                                    <div class="orderSummaryTop">
                                        <h4 class="orderTitle">Sipariş Özeti</h4>
                                        <div class="orderQuantity">1 Ürün</div>
                                        <div class="priceCell">
                                            <span class="label">Ara Tutar (KDV Dahil)</span>
                                            <span class="total">{{basketitems[0].price}} TL</span>
                                        </div>
                                        <div class="btnHolder">
                                            <button id="checkout" name="checkout" value="checkout" class="actionBtn">SATIN AL</button>
                                        </div>
                                    </div>
                                    <!-- orderSummaryTop End -->
                                    <div class="orderSummaryBottom">
                                        <div class="summaryRow">
                                            <span class="txt">Ara Toplam</span>
                                            <span class="txt">{{basketitems[0].price}} TL</span>
                                        </div>
                                    </div>
                                    <!-- orderSummaryBottom End -->
                                </div>
                                <!-- checkoutCartDetails end -->
                            </div>
                            <!--**** basketRightFrame End ****-->
                        </div>
                        <!-- basketContentFrame End -->
                    </div>
                </div>
</template>

<script>
  export default {
    name: 'BasketPage',
    data(){
        return{
            count:1,
            basketitems:[
                {urunid:1,img:"https://i.dr.com.tr/cache/82x82-0/originals/0001889645001-1.jpg",name:"Son Cüret",author_name:"Yılmaz Özdil",discount:40,price_w_d:42.00,price:25.20},
                //{urunid:2,img:"https://i.dr.com.tr/cache/82x82-0/originals/0001893554001-1.jpg",name:"Ev",author_name:"Nermin Yıldırım",discount:35,price_w_d:27.00,price:7.90}
        ]
    }
    },
    methods:{
        Decrase(){
            if(this.count>1)
            {
                this.count--;
            }
        }
    }
    //pnumber:this.$BasketItems.getters.basketitems.length
  }
</script>

<style scoped lang="scss">
@import "../css/dynamic-banner.css";
@import "../css/dr-custom.css";
@import "../css/style.css";
@import "../css/cartSlickSlider.css";
@import "../css/basket-page.css";
@import "../css/form-ui.css";
@import "../css/order-summary.css";


</style>