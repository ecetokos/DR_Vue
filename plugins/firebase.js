/*import * as firebase from 'firebase/app';
import 'firebase/auth'*/
import firebase from '@firebase/app';
require('firebase/auth');
/*import firebase from 'firebase'
require('firebase/auth')*/
/*
<script src="https://www.gstatic.com/firebasejs/8.2.1/firebase-app.js"></script>
<script>
  // Your web app's Firebase configuration
  
  // Initialize Firebase
  
</script>*/

var firebaseConfig = {
    apiKey: "AIzaSyD0c9a38G_8L7k09tqHmrmU2Hk_CeUBhEQ",
    authDomain: "dr-firebase-9de85.firebaseapp.com",
    projectId: "dr-firebase-9de85",
    storageBucket: "dr-firebase-9de85.appspot.com",
    messagingSenderId: "602632802158",
    appId: "1:602632802158:web:f59e6d568054f7f743fb7f"
  };
  
  firebase.initializeApp(firebaseConfig);

/*let app =null;
if(!firebase.apps.length){
    app=firebase.initializeApp(firebaseConfig);
}
export default firebase;*/