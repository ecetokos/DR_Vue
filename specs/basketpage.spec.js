import { mount,shallowMount } from '@vue/test-utils'
import NavBar from '../src/components/Navbar.vue'
import Basket from '../src/components/Basket.vue'
import Store from '../store/index'
import Vue from 'vue'
import Vuex from 'vuex'
import * as $fire from 'firebase';
Vue.use(Vuex)

test('are basket products loaded', () => {
    const store=new Vuex.Store(Store)
    const wrapperb=mount(Basket,{store})
    store.dispatch('signIn',{email:'nisademir@gmail.com',password:'Nisademir1'})
    var user=store.state.user
    var cart=[]
    var quantitytotal=0
    var totalquantityy=0
    var totalprice=0
    var useruid='iXedtLz6GuO16hoXPhHg9WqYZQD3'
    $fire.database.ref('/users/'+useruid+'/basketitems').on('value',(snapshot)=>{      
        cart=snapshot.val()
        quantitytotal=Object.keys(cart).length
        if(quantitytotal>0){
            snapshot.forEach((element) => {
                sayi+=element.val().quantity
                toplam+=element.val().quantity*element.val().price          
            })
            totalquantityy=sayi
            totalprice=toplam
        }
      })  
        var bptotalq=wrapperb.find('#totalquantity').value
        var bptotalp=wrapperb.find('#totalprice').value
        var linumber=wrapperb.find('#products').lastChild.find('#index').innerHTML
        expect(bptotalq).toEqual(totalquantityy)
        expect(bptotalp).toEqual(totalprice)
        expect(linumber).toEqual(quantitytotal)  
})

test('is basketpage vueinstance',()=>{
    const store=new Vuex.Store(Store)
    const wrappernb=mount(Basket,{store})
    expect(wrappernb.isVueInstance()).toBe(true)
})

test('are tages true',()=>{
    const store=new Vuex.Store(Store)
    const wrappernb=mount(Basket,{store})
    const span=wrappernb.find('span')
    expect(span.name()).toBe('span')
    const label=wrappernb.find('label')
    expect(label.name()).toBe('label')
    const input=wrappernb.find('input')
    expect(input.name()).toBe('input')
})

test('can product remove from basket',()=>{
    const store=new Vuex.Store(Store)
    const wrapperb=mount(Basket,{store})
    var cartItem={}
    //productId=5
    store.dispatch('signIn',{email:'nisademir@gmail.com',password:'Nisademir1'})
    var user=store.state.user
    this.$fire.database.ref('/users/'+user.uid+'/basketitems/'+5).on('value',(snapshot)=>{      
        cartItem=snapshot.val()})
    store.dispatch('removeproduct',{id:5,user:user})
    expect(cartItem).toBe(false)
})

test('does basket page contain tages',()=>{
    const store=new Vuex.Store(Store)
    const wrappernb=mount(Basket,{store})
    expect(wrappernb.contains('div')).toBe(true)
    expect(wrappernb.contains('label')).toBe(true)
    expect(wrappernb.contains('span')).toBe(true)
    expect(wrappernb.contains('a')).toBe(true)
    expect(wrappernb.contains('h1')).toBe(true)
    expect(wrappernb.contains('button')).toBe(true)
})

test('has the expected html structure',()=>{
    const store=new Vuex.Store(Store)
    const wrapperbp=mount(Basket,{store})
    expect(wrapperbp.element).toMatchSnapshot()
})

describe('Basket',()=>{
    it('has functions',()=>{
        expect(typeof Basket.methods.Decrase).toBe('function')
        expect(typeof Basket.methods.urunusil).toBe('function')
        expect(typeof Basket.data).toBe('function')
    })
})