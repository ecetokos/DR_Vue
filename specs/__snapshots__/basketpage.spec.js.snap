// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has the expected html structure 1`] = `
<div
  class="container"
  style="margin-top: 40px; width: 100%;"
>
  <div
    class="row"
  >
    <div
      class="container"
    />
     
    <div
      class="col-xs-12 subHeader blackColor"
    >
      <h1>
         Sepetim 
      </h1>
       
      <span>
        ( 
        <span
          id="totalquantity"
        >
          0
        </span>
         Ürün )
      </span>
    </div>
     
    <div
      class="basketContentFrame"
    >
      <div
        class="basketLeftFrame col-md-7 col-lg-8"
        style="float: left; width: 750px;"
      >
        <div
          class="basketProductSummary"
        >
          <div
            class="basketHead row col-hidden"
          >
            <div
              class="col-1 col-sm-1"
            >
              <label
                class="control control-checkbox"
              >
                <input
                  class="newChk"
                  id="checkall"
                  type="checkbox"
                />
                 
                <span
                  class="control-indicator"
                />
              </label>
            </div>
             
            <div
              class="col-11 col-sm-5"
            >
              ÜRÜN 
            </div>
             
            <div
              class="col-sm-1"
            >
                 
            </div>
             
            <div
              class="col-sm-2"
            >
              ADET 
            </div>
             
            <div
              class="col-sm-3 text-right"
            >
              FİYAT 
            </div>
          </div>
           
          <ul
            class="urunler"
          />
        </div>
         
        <div
          class="giftPackage"
        >
          <div
            class="chooseRow"
          >
            <img
              height="20px"
              src="https://www.flaticon.com/svg/static/icons/svg/1161/1161667.svg"
              style="margin-right: 15px;"
              width="20px"
            />
             
            <span
              class="gift-text"
            >
               Hediye Paketi İstiyorum
            </span>
             
            <span
              class="editGiftText"
            >
              Düzenle
            </span>
             
            <span
              class="removeGiftText"
              id="removeGiftText"
            >
              Sil
            </span>
             
            <div
              class="editableGiftText"
            >
              <strong>
                Hediye Notu:
              </strong>
               
              <span
                id="displayGiftMessage"
              />
            </div>
          </div>
        </div>
         
        <div
          class="totalActions col-hidden"
        >
          <a
            class="totalRemove"
            href="javascript:;"
          >
             Sil
          </a>
           
          <a
            class="totalAddFav"
            href="javascript:;"
          >
             Favorilerime Ekle
          </a>
        </div>
      </div>
       
      <div
        class="basketRightFrame col-md-5 col-lg-4"
        style="width: 400px; height: 300px;"
      >
        <div
          class="checkoutCartDetails"
        >
          <div
            class="orderSummaryTop"
          >
            <h4
              class="orderTitle"
            >
              Sipariş Özeti
            </h4>
             
            <div
              class="orderQuantity"
            >
              0 Ürün
            </div>
             
            <div
              class="priceCell"
            >
              <span
                class="label"
              >
                Ara Tutar (KDV Dahil)
              </span>
               
              <span
                class="total"
                id="totalprice"
              >
                0.00 TL
              </span>
            </div>
             
            <div
              class="btnHolder"
            >
              <button
                class="actionBtn"
                id="checkout"
                name="checkout"
                value="checkout"
              >
                SATIN AL
              </button>
            </div>
          </div>
           
          <div
            class="orderSummaryBottom"
          >
            <div
              class="summaryRow"
            >
              <span
                class="txt"
              >
                Ara Toplam
              </span>
               
              <span
                class="txt"
              >
                0.00 TL
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <div>
    
  </div>
</div>
`;
